impl/asm-based-transformer/src/main/java/org/wildfly/transformer/asm/TransformerImpl.java

```diff
@@ -423,8 +421,14 @@ public void clearTransformationState() {
 
     @Override
     public Resource transform(final Resource r) {
+        
+        clearTransformationState(); // clear transformation state for each resource, prior to transformation
+        
         String oldResourceName = r.getName();
         String newResourceName = replacePackageName(oldResourceName, false);
+        if (!newResourceName.equals(oldResourceName)) {  // any file rename counts as a transformation 
+            setClassTransformed(true);
```

Comment:
This is not sufficient as constant pool isn't updated properly.
If you repackage the class only the classloading process will fail.
Here's simple reproducer:
---
public class E {
    public static void main(String[] args) throws Exception {
	final String s1 = "jakarta.servlet.SingleThreadModel";
	Class c = Class.forName(s1);
	System.out.println(c);
    }
}
---
Exception thrown:
---
[blackhole][/home/opalka]>java -cp jakarta-servlet-api-2.3-asm-transformer.jar:. E
Exception in thread "main" java.lang.NoClassDefFoundError: jakarta/servlet/SingleThreadModel (wrong name: javax/servlet/SingleThreadModel)
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at E.main(E.java:5)
---
---